### ISSUE TITLE
Site_url vs dev_addr

### ISSUE THREAD

**Issue by NimaSaed** [squidfunk/mkdocs-material#731]
If you set dev_addr in yml file, it is going to overwrite the site_url value.

```
dev_addr: 0.0.0.0:8000
site_url: http://domain.com/
```

**Comment by squidfunk** [comment]
This is an issue with MkDocs, not Material. Please create an issue on their tracker.

**Comment by NimaSaed** [comment]
@squidfunk, I think the problem is something else. I'm talking about class="md-header-nav__button md-logo", you are using dev_addr to set the href value, if you leave that value empty, then it follows URL host value, mkdocs does this.

**Comment by squidfunk** [comment]
I personally think this is correct behavior, as you want to develop locally and so all reference to `site_url` must be replaced by `dev_addr`, as there may be assets prefixed with the `site_url` within the template which would break when serving locally. One could argue that this is what `base_url` is for, so maybe `site_url` is overridden accidentally (or just for convenience - don't know). Did you ask on the MkDocs issue tracker?

**Comment by squidfunk** [comment]
Ah, I remember. You're right. I changed it a long time ago from `nav.homepage.url` to `site_url` as the former doesn't (or didn't, don't know if they fixed it) work on the 404 error page. Maybe we should check in with that again.

**Comment by NimaSaed** [comment]
How about go with priority of 1.`site_url`, 2.`dev_addr`, 3.`nav.homepage.url`?
Now I remember why I set both `dev_addr` and `site_url`. I set `0.0.0.0:portNum` for each MKDocs on same server without opening all those ports, then I use Apache load balance on `port 80` to point to those MKDocs instances base on the URL request. I set `site_url` because I wanted users click on logo and come back to landing page.
